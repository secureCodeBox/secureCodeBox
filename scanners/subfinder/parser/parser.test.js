// SPDX-FileCopyrightText: the secureCodeBox authors
//
// SPDX-License-Identifier: Apache-2.0

const { readFile } = require("fs/promises");
const util = require("util");

const {
  validateParser,
} = require("@securecodebox/parser-sdk-nodejs/parser-utils");

const { parse } = require("./parser");
const exp = require("node:constants");

test("should properly parse subfinder json file without ip output", async () => {
  const fileContent = await readFile(
    __dirname + "/__testFiles__/passive_scan_without_ip_example.com.jsonl",
    {
      encoding: "utf8",
    },
  );
  const findings = await parse(fileContent);
  // validate findings
  await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot();
});

test("should properly parse subfinder json file with ip output", async () => {
  const fileContent = await readFile(
    __dirname + "/__testFiles__/active_scan_with_ip_example.com.jsonl",
    {
      encoding: "utf8",
    },
  );
  const findings = await parse(fileContent);
  // validate findings
  await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot()
});

test("should properly parse empty json file", async () => {
  const fileContent = await readFile(__dirname + "/__testFiles__/empty.jsonl", {
    encoding: "utf8",
  });
  const findings = await parse(fileContent);
  // validate findings
  await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot()
});
