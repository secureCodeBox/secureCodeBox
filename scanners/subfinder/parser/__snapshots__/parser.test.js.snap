// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should properly parse empty json file 1`] = `[]`;

exports[`should properly parse subfinder json file with ip output 1`] = `
[
  {
    "attributes": {
      "addresses": {
        "ip": "2.16.206.21",
      },
      "domain": "example.com",
      "hostname": "www.example.com",
      "ip_address": "2.16.206.21",
      "source": "crtsh",
    },
    "category": "Subdomain",
    "description": "Found subdomain www.example.com",
    "identified_at": null,
    "location": "www.example.com",
    "name": "www.example.com",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
]
`;

exports[`should properly parse subfinder json file without ip output 1`] = `
[
  {
    "attributes": {
      "addresses": {
        "ip": null,
      },
      "domain": "example.com",
      "hostname": "releases.example.com",
      "ip_address": null,
      "source": "alienvault",
    },
    "category": "Subdomain",
    "description": "Found subdomain releases.example.com",
    "identified_at": null,
    "location": "releases.example.com",
    "name": "releases.example.com",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": {
        "ip": null,
      },
      "domain": "example.com",
      "hostname": "myap.example.com",
      "ip_address": null,
      "source": "alienvault",
    },
    "category": "Subdomain",
    "description": "Found subdomain myap.example.com",
    "identified_at": null,
    "location": "myap.example.com",
    "name": "myap.example.com",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": {
        "ip": null,
      },
      "domain": "example.com",
      "hostname": "not-exist.example.com",
      "ip_address": null,
      "source": "alienvault",
    },
    "category": "Subdomain",
    "description": "Found subdomain not-exist.example.com",
    "identified_at": null,
    "location": "not-exist.example.com",
    "name": "not-exist.example.com",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": {
        "ip": null,
      },
      "domain": "example.com",
      "hostname": "new-client.example.com",
      "ip_address": null,
      "source": "alienvault",
    },
    "category": "Subdomain",
    "description": "Found subdomain new-client.example.com",
    "identified_at": null,
    "location": "new-client.example.com",
    "name": "new-client.example.com",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": {
        "ip": null,
      },
      "domain": "example.com",
      "hostname": "op.example.com",
      "ip_address": null,
      "source": "alienvault",
    },
    "category": "Subdomain",
    "description": "Found subdomain op.example.com",
    "identified_at": null,
    "location": "op.example.com",
    "name": "op.example.com",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": {
        "ip": null,
      },
      "domain": "example.com",
      "hostname": "mall.example.com",
      "ip_address": null,
      "source": "alienvault",
    },
    "category": "Subdomain",
    "description": "Found subdomain mall.example.com",
    "identified_at": null,
    "location": "mall.example.com",
    "name": "mall.example.com",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": {
        "ip": null,
      },
      "domain": "example.com",
      "hostname": "cdn1.example.com",
      "ip_address": null,
      "source": "alienvault",
    },
    "category": "Subdomain",
    "description": "Found subdomain cdn1.example.com",
    "identified_at": null,
    "location": "cdn1.example.com",
    "name": "cdn1.example.com",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": {
        "ip": null,
      },
      "domain": "example.com",
      "hostname": "xn--rksmrgs-5wao1o.example.com",
      "ip_address": null,
      "source": "alienvault",
    },
    "category": "Subdomain",
    "description": "Found subdomain xn--rksmrgs-5wao1o.example.com",
    "identified_at": null,
    "location": "xn--rksmrgs-5wao1o.example.com",
    "name": "xn--rksmrgs-5wao1o.example.com",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
]
`;
